### DISCUSSION TITLE
kiali dashboard throughs "Could not fetch health: Error while fetching app health: client_error: client error: 404"

### DISCUSSION THREAD

**Discussion by Udhay-CIT** [kiali/kiali#5968] [Q&A] [answered]
**Setup**
I have kiali installed in EKS cluster, istio and prometheus are already in place. 
kiali-operator - 1.65
kiali                 - 1.65
kiali and prometheus running in observabiliy namespace, istiod running in istio-system namespace
Kiali integration to external services like istio and prometheus are proper as per KIALI CR documentation.

**Issue**
1. In the dashboard I could see the "list of Istio Config", "list of workloads" but if I go inside the workloads it says "No Workload is selected".
2. I could see a popup error on right top cornor saying "Could not fetch health: Error while fetching app health: client_error: client error: 404"
3. Graph is not visible, it says "Cannot load the graph"

**Pod logs** 
"Error fetching Health in namespace <namespace> for workload <workload>: client_error: client error: 404"

**Comment by jshaughn** [comment]
Not sure, sounds environmental like Kiali can't reach Kube API.

**Reply by Udhay-CIT** [reply]
![image](https://user-images.githubusercontent.com/126764227/228850951-5ff76efe-4ca9-48b5-a768-30f64767b30a.png)

the above attached is the kiali pod logs.
external_services:
          prometheus:
            url: "http://kube-prometheus-stack-prometheus.observability:9090"
            
this is the only prometheus config I have in kiali values.yaml.
Is there any more configuration needed for this integration?

**Reply by jmazzitelli** [reply]
404 means the prometheus url you gave it is not correct.... that means page not found - the prometheus url is bad.

Review the docs on this to see if it will help: https://kiali.io/docs/configuration/p8s-jaeger-grafana/prometheus/

and this provides some docs on all the prometheus settings: 

https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.custom_dashboards.prometheus

**Reply by Udhay-CIT** [reply]
Thanks for the docs link and comment here.

I changed the prometheus url to "http://kube-prometheus-stack-prometheus.observability:9090/prometheus/graph" because this was the which gives "200 OK" response when I try to hit from insde the kiali container.
but even after changing it to correct url, the "client error 404" remains.

**Comment by jmazzitelli** [comment]
you don't want to go to the /graph URL - that's the graphical user interface endpoint. You want to go to the api endpoint - Kiali queries the api endpoint, not the UI endpoint

**Reply by Udhay-CIT** [reply]
you are right. 
prometheus endpoint in our case is "http://kube-prometheus-stack-prometheus.observability:9090/prometheus"
changing to this worked. Thanks for the support

**Comment by Udhay-CIT** [comment, marked as answer]
using right prometheus internal endpoint helped
